components/sp-solutions/org.wso2.carbon.sp.solutions.apim.analytics/components/org.wso2.sp.extension.siddhi.io.mgwfile/src/main/java/org/wso2/extension/siddhi/io/mgwfile/util/FileDataRetrieverUtil.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.extension.siddhi.io.mgwfile.util;
+
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.wso2.carbon.databridge.commons.AttributeType;
+import org.wso2.carbon.databridge.commons.StreamDefinition;
+import org.wso2.extension.siddhi.io.mgwfile.MGWFileSourceConstants;
+import org.wso2.extension.siddhi.io.mgwfile.exception.MGWFileSourceException;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Util Class for MGWFileDataRetriever
+ */
+public class FileDataRetrieverUtil {
+
+    private static volatile Map<String, JSONArray> streamDefinitions = new HashMap<>();
+
+    public static void addStreamDefinition(StreamDefinition streamDefinition, String streamId) throws
+            MGWFileSourceException {
+        try {
+            String jsonStr = streamDefinition.toString();
+            JSONParser jsonParser = new JSONParser();
+            JSONObject jsonObject = (JSONObject) jsonParser.parse(jsonStr);
+            streamDefinitions.put(streamId, (JSONArray) jsonObject.get("payloadData"));
+        } catch (ParseException e) {
+            throw new MGWFileSourceException("Error during parsing stream definition", e);
+        }
+    }
+
```

Comment:
remove extra space

Reply:
Fixed in a21aa920ceb5fdbd3f34fa9dddb75b5630876048 of https://github.com/wso2/analytics-apim.